### PR TITLE
chore(INT-562): update component test suite

### PR DESCRIPTION
**Pull request by Sanjay-Veernala**
## Description of the change
Resolved INT-562
> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: testScript.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+const fs = require("fs");
```
Comment (sanpj2292) [inline testScript.js:1]: Do we need this file ?

File: testScript_1.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
+const fs = require('fs');
```
Comment (sanpj2292) [inline testScript_1.js:1]: Do we need this file ?

File: package-lock.json [hunk 3]
```diff
@@ -23,7 +23,6 @@
         "ajv": "^8.12.0",
```
Comment (sanpj2292) [inline package-lock.json:23]: Why do we have changes in this file ?

File: testScript.js (hunk 1 above)
Comment (Sanjay-Veernala) [reply testScript.js:1]: No, 
we donot need this
removing the file now.

**Comment by sanpj2292** [comment]
- For all the destinations `test.js` are not removed.
- `vero.json` is not removed.

Can you also fix the conflict and make sure the tests are successful ?

**Review by utsabc — DISMISSED** [review]

**Review by utsabc — APPROVED** [review]

File: test/integrations/destinations/snapchat_conversion/processor/data.ts [hunk 4]
```diff
@@ -0,0 +1,4300 @@
...
+                },
+                screen: {
+                  density: 2,
+                },
+                externalId: [
+                  {
+                    type: 'ga4AppInstanceId',
+                    id: 'f0dd99v4f979fb997ce453373900f891',
+                  },
+                ],
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/destinations/snapchat_conversion/processor/data.ts:53]: https://github.com/rudderlabs/rudder-transformer/blob/develop/test/__tests__/data/snapchat_conversion.json#L43-L44

Here we can see that the same information is present. We might need to clean-up such data-related issues later

File: test/integrations/destinations/snapchat_conversion/router/data.ts [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
..."event_type":"SEARCH","user_agent":"mozilla/5.0 (macintosh; intel mac os x 10_15_2) applewebkit/537.36 (khtml, like gecko) chrome/79.0.3945.88 safari/537.36","hashed_email":"73062d872926c2a556f17b36f50e328ddf9bff9d403939bd14b6c3b7f5a33fc2","hashed_phone_number":"bc77d64d7045fe44795ed926df37231a0cfb6ec6b74588c512790e9f143cc492","hashed_idfv":"54bd0b26a3d39dad90f5149db49b9fd9ba885f8e35d1d94cae69273f5e657b9f","hashed_mobile_ad_id":"f9779d734aaee50f16ee0011260bae7048f1d9a128c62b6a661077875701edd2","timestamp":"1650625078","event_conversion_type":"OFFLINE","pixel_id":"dummyPixelId"},{"search_string":"t-shirts","event_type":"SEARCH","user_agent":"mozilla/5.0 (macintosh; intel mac os x 10_15_2) applewebkit/537.36 (khtml, like gecko) chrome/79.0.3945.88 safari/537.36","hashed_email":"73062d872926c2a556f17b36f50e328ddf9bff9d403939bd14b6c3b7f5a33fc2","hashed_phone_number":"bc77d64d7045fe44795ed926df37231a0cfb6ec6b74588c512790e9f143cc492","hashed_idfv":"54bd0b26a3d39dad90f5149db49b9fd9ba885f8e35d1d94cae69273f5e657b9f","hashed_mobile_ad_id":"f9779d734aaee50f16ee0011260bae7048f1d9a128c62b6a661077875701edd2","timestamp":"1650625078","event_conversion_type":"OFFLINE","pixel_id":"dummyPixelId"}]',
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/destinations/snapchat_conversion/router/data.ts:237]: It is a string, don't think better formatting would be possible directly. But if we want to really try, we can do `JSON.stringify({})`, only thing is doing in this PR is a little time-taking. Let's keep this in mind while we are planning to clean-up test-case data

File: test/integrations/destinations/snapchat_conversion/processor/data.ts [hunk 6]
```diff
@@ -0,0 +1,4300 @@
...
+            },
+            statusCode: 400,
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'snapchat_conversion',
+    description: 'Test case where snap app id is not present in destination config',
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/destinations/snapchat_conversion/processor/data.ts:887]: snap app id & appId are two different fields. One case is test-case to validate presence of `appId` & this case is to validate presence of `snapAppId`. So this is not duplicate

File: test/integrations/destinations/snapchat_custom_audience/dataDelivery/data.ts [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    "output": {
+      "response": {
+        "status": 500,
+        "body": {
+          "output": {
+            "status": 500,
+            "destinationResponse": {
+              "response": "unauthorized",
+              "status": 401
+            },
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/destinations/snapchat_custom_audience/dataDelivery/data.ts:126]: Not sure what you mean here. Can you elaborate ?

**Review by ItsSudip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Comment by utsabc** [comment]
LGTM from my side comments are not blocking feel free to resolve

**Review by sanpj2292 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**